core/src/main/scala/akka/kafka/scaladsl/Committer.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka.scaladsl
+import akka.Done
+import akka.kafka.CommitterSettings
+import akka.kafka.ConsumerMessage.{Committable, CommittableOffset, CommittableOffsetBatch}
+import akka.stream.scaladsl.{Flow, Keep, Sink}
+
+import scala.concurrent.Future
+
+object Committer {
+
+  def sink(settings: CommitterSettings): Sink[Committable, Future[Done]] =
+    Flow[Committable]
+    // Not very efficient, ideally we should merge offsets instead of grouping them
+      .groupedWeightedWithin(settings.maxBatch, settings.interval)(commitableSize)
+      .map(_.foldLeft(CommittableOffsetBatch.empty)(_ updated _))
```

Comment:
We should push that logic down to `CommittableOffsetBatch`, there is an `apply` method for `Seq[CommittableOffset]`. I guess it can be changed to accept `Committable` instead.

Reply:
Thanks, I haven't noticed that there is such method.